tools/pylib/boututils/showdata.py

```diff
@@ -325,6 +325,11 @@ def showdata(vars, titles=[], legendlabels = [], surf = [], polar = [], tslice =
 
     # Generate grids for plotting
     # Try to use provided grids where possible 
+    # If x and/or y are not lists, apply to all variables
+    if not isinstance(x, (list,tuple)):
```

Comment:
What happens if `x` is a numpy `ndarray`? We probably want all the `isinstance(..., list)` checks to be `isinstance(..., (list, tuple, np.ndarray))`

Reply:
The use case I have in mind is that x=Rxy and y=Zxy so that showdata will make contour plots in the R-Z domain like this:
![animation](https://user-images.githubusercontent.com/3958036/35915695-c4dafdea-0bff-11e8-9eae-9e5526cb3b6d.gif)
So my aim was to check if we have a single ndarray, or different ones for different variables. It will fail if you try to pass in an 'array of arrays', so more checking the sanity of inputs would be useful.

I think it's too hard to tell the difference between a 2d array for the x-coordinate and a 2d array of 1d coordinates for each variable, so if the user wants a different x and y for each variable, we'll have to make them pass a list not an array, but I will try to add some sanity checking with helpful error messages if dimensions don't match, etc.